### DISCUSSION TITLE
Build failed from source

### DISCUSSION THREAD

**Discussion by AryanJ13** [NVIDIA/cuda-quantum#996] [Q&A] [answered]
Hello, 
I am trying to build cuda-quantum inside a docker container as explained in contributor guidelines(i am using all default settings) when i am running the build script it is failing with the Exit Code 1. is it due to the lack of memory in docker container? my laptop only has 16 GB of ram so i am not able to allocate more than that to wsl. is there any way to get around it using swap memory?  please tell me a way to fix this.

Thanks

**Comment by boschmitt** [comment]
Greetings,

It would be hard to know exactly what the problem is without more information. Based on what you said, I believe the problem might actually lie on lack of storage space: compiling cuda-quantum from source requires lots of it. Try allowing docker to use more space for containers.

**Comment by bmhowe23** [comment]
Also, could you please clarify whether or you not you are building LLVM as part of the process? I know that the LLVM build consumes a lot of RAM, and the exact amount varies as a function of the number of parallel threads you allow the build system to use.

If you are using the `scripts/build_cudaq.sh` script, you can try running with `-v` to get additional information printed to the console, though this will only have an effect on the cuda-quantum portion of the build, not the LLVM portion of the build.

**Comment by wszdexdrf** [comment]
There seemed to be a problem with the submodules in tpls/*. Manual cloning with `git clone --recursive` and setting the option `GIT_SUBMODULE` to OFF fixed that issue. Unfortunately, for me I got a `[ 3366.888245] Out of memory: Killed process 28653 (cc1plus)` error from dmesg, so I couldn't test further.

**Comment by bmhowe23** [comment, marked as answer]
I have a hunch that it is LLVM (as LLVM is the biggest one to compile in `tpls`). **Are you able to change the docker container you're using?** If so, the `ghcr.io/nvidia/cuda-quantum-devdeps:ext-gcc12-main` container (built regularly by our CI) has LLVM already pre-built, so you wouldn't have to build LLVM from scratch if that is an option. That would be best in my opinion, but if that doesn't work, let me know. I've built LLVM on my laptop w/ 16 GB of RAM a while back, but it takes a bit of manual tinkering with the `-j` option (and trial and error) so that the number of worker threads is limited, which obviously slows the build down. It's a little painful, but it I believe it is technically possible.

**Reply by AryanJ13** [reply]
Thank you @bmhowe23 adding -j2 to ninja install worked for me!
